r/tools/winlibs.R

```diff
@@ -53,7 +53,7 @@ if (!file.exists(sprintf("windows/arrow-%s/include/arrow/api.h", VERSION))) {
     dev_version <- package_version(VERSION)[1, 4]
 
     # Small dev versions are added for R-only changes during CRAN submission.
-    if (is.na(dev_version) || dev_version < 100) {
+    if (is.na(dev_version) || dev_version < "100") {
```

Comment:
Does the logic still work? gt/lt string comparisons can be funky iirc. Maybe we should explicitly convert it to a version object (or what ever it's called)

Reply:
This is what was recommended in the email from CRAN, but you're right, i.e. `"10" > "2" == FALSE`.

Reply:
Actually, nope, it doesn't matter; because `dev_version` is a `package_version` object, it's fine (i.e. in this case testing that a dev version of `10` is higher than the string "2"  returns TRUE in the above comparison when done with the right classes)

Reply:
``` r
x <- package_version("0.0.0.10")[1, 4]
x
#> [1] '10'
x > 2
#> [1] TRUE
```

<sup>Created on 2023-07-27 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
Ah nice, yay for implicit casts :D